pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/forward/ForwardIndexReaderFactory.java

```diff
@@ -81,13 +81,17 @@ public static ForwardIndexReader createRawIndexReader(PinotDataBuffer dataBuffer
             ? new FixedBytePower2ChunkSVForwardIndexReader(dataBuffer, storedType)
             : new FixedByteChunkSVForwardIndexReader(dataBuffer, storedType);
       } else {
-        return version == VarByteChunkForwardIndexWriterV4.VERSION ? new VarByteChunkSVForwardIndexReaderV4(dataBuffer,
-            storedType) : new VarByteChunkSVForwardIndexReader(dataBuffer, storedType);
+        return version == VarByteChunkForwardIndexWriterV4.VERSION ? new VarByteChunkForwardIndexReaderV4(
```

Comment:
Consider reversing the check to first check the version

Reply:
This logic is quite convoluted. I've updated to make it shorter, please have a look. We may think about refactoring VarByteChunkSVForwardIndexReader and VarByteChunkMVForwardIndexReader to be a single class similar to the new v4 reader.